File: kotlin/src/main/kotlin/com/di/kotlin/no/di/UserRepository.kt
package com.di.kotlin.no.di

import com.di.kotlin.model.User

class UserRepository {
    private val logger = Logger()

    private val users: MutableList<User> = mutableListOf()

    fun findById(id: Long) : User? {
        return users.find { it.id == id }
    }

    fun createUser(nombre: String, apellido: String) : User{
        this.logger.log("user con nombre $nombre y apellido $apellido creado")
        val user = User(Math.random().toLong(), nombre, apellido)
        users.add(user)
        return user
    }
}

File: nodejs/src/sinDependencyInjection/UserRepository.js
import {User} from "./User";
import {Logger} from "./Logger";

export class UserRepository {
    users = []
    logger = new Logger()

    create(nombre, apellido) {
        const user = new User(Math.random(), nombre, apellido)
        this.users.push(user);
        this.logger.log(`user con nombre ${nombre} y apellido ${apellido} creado`)
        return user
    }

    findById(id) {
        return this.users.find(user => user.id === id)
    }
}